server/src/server.ts

```diff
@@ -800,32 +814,24 @@ connection.onTypeDefinition(async (typeDefinitionParams) => {
 })
 
 connection.onReferences(async (referencesParams) => {
-	const doc = documents.get(referencesParams.textDocument.uri)
-	if (!doc)
-		return null
-	const fileData = await getDocumentData(referencesParams.textDocument.uri)
-	if (!fileData)
-		return null
-	const callChain = findCallChain(doc, fileData, referencesParams.position, /*forAutocompletion*/false)
-	if (callChain.length === 0)
-		return null
-	const last = callChain[callChain.length - 1]
-	const foundTokens = last.tokens
-
+	const [fileData, doc, callChain] = await getTokenData(referencesParams.textDocument.uri, referencesParams.position)
```

Comment:
what should happen here when getTokenData returns null?

i think getTokenData is completely unnecessary function which makes code more complicated and incorrect